Sources/FluentKit/Query/QueryBuilder.swift

```diff
@@ -276,11 +296,27 @@ public final class QueryBuilder<Model>
             filter.value
         )
     }
+
+    @discardableResult
+    public func filter<Joined>(_ alias: Joined.Type, _ filter: ModelFieldFilter<Joined, Joined>) -> Self
+        where Joined: FluentKit.Model
+    {
+        return self.filter(
+            .field(path: filter.lhsPath, schema: Joined.schema, alias: nil),
+            filter.method,
+            .field(path: filter.rhsPath, schema: Joined.schema, alias: nil)
+        )
+    }
     
     @discardableResult
     public func filter<Value>(_ field: KeyPath<Model, Field<Value>>, _ method: DatabaseQuery.Filter.Method, _ value: Value) -> Self {
         return self.filter(Model.key(for: field), method, value)
     }
+
+    @discardableResult
+    public func filter<Value>(_ lhsField: KeyPath<Model, Field<Value>>, _ method: DatabaseQuery.Filter.Method, _ rhsField: KeyPath<Model, Field<Value>>) -> Self {
+        return self.filter(Model.key(for: lhsField), method, Model.key(for: rhsField))
+    }
```

Comment:
This is now defined with regard to the generic `Model` of `QueryBuilder` is that the correct way to do it?